### PR TITLE
Fix race condition in `LDBDatabase.Close()`

### PR DESCRIPTION
**Pull request by pedropombeiro**
- https://jenkins.status.im/job/status-go/job/race-check/54/consoleFull

This patch fixes the race condition discovered in the build above.

<blockquote></blockquote>

### CODE REVIEW

**Review by dshulyak — CHANGES_REQUESTED** [review]
sorry, that is my mistake. there should be a defer on L176 (can't comment in code) then db.quitChan will be written only when part that reads it is already exited

**Comment by pedropombeiro** [comment]
Ah, makes sense. Pushed new commit.

**Comment by pedropombeiro** [comment]
@dshulyak Is there an upstream PR that needs to be updated?

File: _assets/patches/geth/0026-ethdb-error-deadlock.patch [hunk 1]
```diff
@@ -28,14 +28,14 @@ index 001d8f0bb..4c9b1412c 100644
  	err := db.db.Close()
  	if err == nil {
  		db.log.Info("Database closed")
-@@ -173,7 +171,11 @@ func (db *LDBDatabase) Meter(prefix string) {
```
Comment (dshulyak) [inline _assets/patches/geth/0026-ethdb-error-deadlock.patch]: strange, why methods were removed?

**Review by dshulyak — APPROVED** [review]
no, i wanted to push one and then forgot.

File: _assets/patches/geth/0026-ethdb-error-deadlock.patch (hunk 1 above)
Comment (pedropombeiro) [reply _assets/patches/geth/0026-ethdb-error-deadlock.patch]: Maybe I'm not using the same flags to generate the diff file... I'm doing `diff -ur`

Comment (pedropombeiro) [reply _assets/patches/geth/0026-ethdb-error-deadlock.patch]: Guess I need to add the `-p` flag

**Comment by dshulyak** [comment]
it looks like that deadlock is back now https://travis-ci.org/status-im/status-go/jobs/374364345

<blockquote><div><strong><a href="https://travis-ci.org/status-im/status-go/jobs/374364345">Travis CI - Test and Deploy Your Code with Confidence</a></strong></div></blockquote>

**Comment by dshulyak** [comment]
😕 there also should be no this line in patch

```
@@ -349,7 +349,6 @@ func (db *LDBDatabase) meter(refresh time.Duration) {
                select {
                case errc := <-db.quitChan:
                        // Quit requesting, stop hammering the database
-                       errc <- nil
                        return
 
                case <-time.After(refresh): 
```

**Comment by pedropombeiro** [comment]
Where do you see that line in the patch?

**Pull request merged**